src/backend/base/langflow/components/nvidia/nemo_evaluator.py

```diff
@@ -382,22 +489,220 @@ def _generate_custom_evaluation_body(self) -> dict:
                 {
                     "eval_type": "automatic",
                     "eval_subtype": "custom_eval",
-                    "input_file": getattr(self, "300_input_file", ""),
+                    "input_file": input_file,
                     "num_of_samples": getattr(self, "350_num_of_samples", -1),
                     "scorers": getattr(self, "351_scorers", ["accuracy", "bleu", "rouge", "em", "bert", "f1"]),
-                    "inference_configs": [
-                        {
-                            "model": {
-                                "llm_name": getattr(self, "000_llm_name", ""),
-                            },
-                            "run_inference": getattr(self, "310_run_inference", True),
-                            "inference_params": inference_params,
-                            "output_file": getattr(self, "320_output_file", "")
-                            if not getattr(self, "310_run_inference", True)
-                            else None,
-                        }
-                    ],
+                    "inference_configs": [inference_config],
+                }
+            ],
+            "tag": getattr(self, "001_tag", ""),
+        }
+
+    async def _generate_llm_as_judge_body(self) -> dict:
+        dataset_path = await self.process_and_upload_dataset()
+        input_file = f"nds:{dataset_path}/input/input.json"
+        # Handle run_inference as a boolean
```

Comment:
I don't understand how run_inference is handled in the code?

Reply:
Custom evaluation can be run in 2 ways. When run-inference is `true` user passes only input file. Evaluator starts the inference model to generates output and evaluates the output. When run_inference is set to `false`, user passes both input and output file, evaluator uses the data in the output file for evaluation.